src/screens/Login/index.tsx

```diff
@@ -1,15 +1,123 @@
+import { useForm, Controller } from 'react-hook-form';
+import { useState } from 'react';
+import { useNavigation } from '@react-navigation/native';
+import { useLoginMutation, AuthUserInput } from '../../generated/graphql';
+import { TouchableOpacity } from 'react-native';
+import * as SecureStore from 'expo-secure-store';
+
+interface FormProps {
+  email: string;
+  password: string;
+}
+
+
+
+const Login: React.FC = () => {
+  const { control, handleSubmit, formState: { errors, isDirty, isValid } } = useForm<FormProps>();
+  const [login, { data, loading, error }] = useLoginMutation();
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+
+  const navigation = useNavigation<any>();
+
+  const emailPattern = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
+  const passwordPattern = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{8,}$/;
+
+
+  const onSubmit = async (data: AuthUserInput) => {
+    try {
+      const res = await login({
+        variables: {
+          input: {
+            email: data.email,
+            password: data.password,
+          },
+        },
+      });
+      console.log(res.data?.login.data, 'data Response') // Data response object
+      
+      const { refreshToken, idToken } = res.data?.login.data // Get tokens from Response
+      
+      if (idToken && refreshToken) {
+        await SecureStore.setItemAsync('idToken', idToken); // Store the tokens in SecureStore       
+        await SecureStore.setItemAsync('refreshToken', refreshToken);
+        setErrorMessage('') // Clear error message
+        navigation.navigate('Home'); // Navigate to the home screen on success
+      }
+    } catch (error) {
+      console.log(error,'rrorrrr');
+      if (error && error.message) {
+        setErrorMessage(error.message); // Set the specific error message for display
+      } else {
+        setErrorMessage('An error occurred. Please try again.'); // Fallback error message
+      }
+    }
+  };
+  const handleRegister = () => {
+    navigation.navigate('Register'); // Replace 'Register' with the name of your registration screen or route
+  };
+  const isFormValid = !errors.email && !errors.password; // Check if both email and password fields have no errors
 
-function Login() {
   return (
-    // <Column safeAreaTop mt={30} h="100%">
-    <Center safeAreaTop mt={30} h="100%">
-      <Heading textAlign="center" mb={6}>
-        Welcome to CraftPile!
-      </Heading>
-    </Center>
-    // </Column>
+      <Center flex={1} py="10">
+        <Box w="90%">
+          <Heading size="lg" fontWeight="600" color="coolGray.800" _dark={{ color: 'warmGray.50' }}>
+            Welcome to CraftPile!
+          </Heading>
+          <Heading mt="1" _dark={{ color: 'warmGray.200' }} color="coolGray.600" fontWeight="medium" size="xs">
+            Sign in to continue!
+          </Heading>
+          <VStack space={3} mt="5">
+            <FormControl isInvalid={!!errors.email?.message}>
```

Comment:
nit (nit pick) - a double bang `!!` coerces a value into a boolean. The safety operator `?.` will return either a value or `undefined` if the chain breaks. So these are a bit redundant together. The following should be sufficient.
```js
<FormControl isInvalid={errors?.email?.message}>
```